tensorflow/core/kernels/diag_op_gpu.cu.cc

```diff
@@ -62,16 +62,10 @@ struct DiagFunctor<GPUDevice, T> {
     const GPUDevice& device = context->eigen_device<GPUDevice>();
     GpuLaunchConfig diag_config =
         GetGpuLaunchConfig(virtual_thread_count, device);
-    TF_CHECK_OK(
-        CudaLaunchKernel(DiagCudaKernel<T>, diag_config.block_count,
-                         diag_config.thread_per_block, 0, device.stream(),
-                         diag_config.virtual_thread_count, size, in, out));
-
-    auto err = cudaGetLastError();
-    if (err != cudaSuccess) {
-      return errors::Internal(
-          "Could not launch DiagOp kernel: ", cudaGetErrorString(err), ".");
-    }
+    TF_CHECK_OK(GpuLaunchKernel(
```

Comment:
TF_RETURN_IF_ERROR would be slightly better.

Reply:
Thanks for letting me know!